ion-schema/src/isl/isl_type.rs

```diff
@@ -217,6 +219,20 @@ impl IslTypeImpl {
     }
 }
 
+impl WriteToIsl for IslTypeImpl {
+    fn write_to<W: IonWriter>(&self, writer: &mut W) -> IonSchemaResult<()> {
+        writer.set_annotations(["type"]);
+        writer.step_in(IonType::Struct)?;
+        writer.set_field_name("name");
+        writer.write_symbol(self.name.clone().unwrap())?;
```

Comment:
In this circumstance, is it guaranteed that `self.name` is always `Some(_)`?

This version will return an error instead of panicking if that were ever to occur:
```rust
self.name.map(|name| writer.write_symbol(name))?;
```